Tests/Tests/Core/Extensions/Foundation/OptionalTests.swift

```diff
@@ -38,4 +38,14 @@ class OptionalTests: XCTestCase {
     let sut: String? = "A"
     XCTAssertFalse(sut.isNilOrEmpty, "Should not return true when string contains characters")
   }
+  
+  func test_replacingString_producesExpectedResult() {
+    let sut: String? = "Nice day today!"
+    XCTAssertEqual(sut.replacingString(with: "night", range: NSRange(location: 5, length: 3)), "Nice night today!")
+  }
+  
+  func test_replacingString_doesNothingOnNilValue() {
+    let sut: String? = nil
+    XCTAssertNotEqual(sut.replacingString(with: "night", range: NSRange(location: 0, length: 0)), "Nice night today!")
```

Comment:
Would it make more sense to make this test asserting for nil value like this:
`XCTAssertNil(sut.replacingString(with: "night", range: NSRange(location: 0, length: 0)))`

Reply:
We can use:
`XCTAssertNotNil(sut.replacingString(with: "night", range: NSRange(location: 0, length: 0)))`
Since `replacingString` returns non-optional String